test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SampleControllerSpec.scala

```diff
@@ -0,0 +1,245 @@
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("change_sample_status-heading")
+    }
+
+    "return 200 OK and HTML content type - For OPEN Case" in {
+      val c = aCase(withStatus(CaseStatus.OPEN), withDecision())
+
+      val result = await(controller(c).chooseStatus("reference")(newFakeGETRequestWithCSRF(fakeApplication)))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("change_sample_status-heading")
+    }
+
+    "return OK when user has right permissions" in {
+      val c = aCase(withStatus(CaseStatus.COMPLETED), withDecision())
+
+      val result = await(controller(c, Set(Permission.EDIT_SAMPLE)).chooseStatus("reference")(newFakeGETRequestWithCSRF(fakeApplication)))
+
+      status(result) shouldBe Status.OK
+    }
+
+    "redirect unauthorised when does not have right permissions" in {
+      val c = aCase(withStatus(CaseStatus.COMPLETED), withDecision())
+
+      val result = await(controller(c, Set.empty).chooseStatus("reference")(newFakeGETRequestWithCSRF(fakeApplication)))
+
+      status(result) shouldBe Status.SEE_OTHER
+      redirectLocation(result).get should include("unauthorized")
+    }
+  }
+
+  "Case Appeal - Submit Status" should {
+
+    "update & redirect - For CANCELLED Case" in {
+      val c = aCase(withReference("reference"), withStatus(CaseStatus.CANCELLED), withDecision())
+
+      given(casesService.updateSampleStatus(refEq(c), any[Option[SampleStatus]], any[Operator])(any[HeaderCarrier])).willReturn(Future.successful(c))
+
+      val result = await(controller(c).updateStatus("reference")(newFakePOSTRequestWithCSRF(fakeApplication).withFormUrlEncodedBody("status" -> "DESTROYED")))
+
+      verify(casesService).updateSampleStatus(refEq(c), refEq(Some(SampleStatus.DESTROYED)), any[Operator])(any[HeaderCarrier])
+
+      status(result) shouldBe Status.SEE_OTHER
+      locationOf(result) shouldBe Some("/tariff-classification/cases/reference/sample")
+    }
+
+    "update & redirect - For COMPLETED Case" in {
+      val c = aCase(withReference("reference"), withStatus(CaseStatus.COMPLETED), withDecision())
+
+      given(casesService.updateSampleStatus(refEq(c), any[Option[SampleStatus]], any[Operator])(any[HeaderCarrier])).willReturn(Future.successful(c))
+
+      val result = await(controller(c).updateStatus("reference")(newFakePOSTRequestWithCSRF(fakeApplication).withFormUrlEncodedBody("status" -> "DESTROYED")))
+
+      verify(casesService).updateSampleStatus(refEq(c), refEq(Some(SampleStatus.DESTROYED)), any[Operator])(any[HeaderCarrier])
+
+      status(result) shouldBe Status.SEE_OTHER
+      locationOf(result) shouldBe Some("/tariff-classification/cases/reference/sample")
+    }
+
+    "update & redirect - For OPEN Case" in {
+      val c = aCase(withReference("reference"), withStatus(CaseStatus.OPEN), withDecision())
```

Comment:
Again - suspect you don't need to test against 3x case status

Reply:
Have updated this one and removed irrelevant repeats